### PR TITLE
service: nfc: Keep tag alive while it's being used

### LINKED ISSUE #6681: Chibi Robo Zip Lash error when trying to read amiibos as of Nightly 1938 or newer
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Nightly 1938 and above

### Description of Issue

Follow-up to issue here:
https://github.com/citra-emu/citra/issues/6667

Issue above was fixed, but game now fails at what I assume is the next step.
When amiibo button is pressed in-game, game now correctly waits for amiibo to be loaded... however after a very short time, the game gives an error message:
"An error occurred while attempting to read amiibo data. Please try again."

For reference, this is the same error I get on my 3dsXL when I (briefly) place an amiibo on the screen and quickly take it away again.

Again, I used Metroid; Samus Returns as an alternate game with amiibo support to test amiibo funcionality in build 1942, and it functions correctly.

I was also able to test amiibo write functionality in the home menu -> theme menu -> amiibo settings -> register owner and nickname
so Citra itself on 1942 is able to write to, and then read that written data from, an amiibo.

### Expected Behavior

In 1936 and earlier, functionality is correct - the game reads the amiibo, and then loads whatever it needs to from it.
Write functionality was not tested on 1936 as it is not supported.

### Reproduction Steps

Load a save of Chibi Robo Zip Lash that has progressed far enough to unlock amiibo support (the first stage).
Press the amiibo button.
Load an amiibo file.

Error will appear.

### Log File

[   0.093160] Input <Info> input_common/udp/client.cpp:StartCommunication:206: Starting communication with UDP input server on 127.0.0.1:26760
[   0.143287] Frontend <Info> citra_qt/main.cpp:GMainWindow:230: Citra Version: Nightly 1942  | HEAD-662bb9b
[   0.143299] Frontend <Info> citra_qt/main.cpp:GMainWindow:245: Host CPU: 12th Gen Intel(R) Core(TM) i5-12400 | AVX2 | FMA
[   0.143355] Frontend <Info> citra_qt/main.cpp:GMainWindow:247: Host OS: Windows 11 Version 22H2 (Build 22621.1848)
[   0.143364] Frontend <Info> citra_qt/main.cpp:GMainWindow:250: Host RAM: 15.79 GiB
[   0.143364] Frontend <Info> citra_qt/main.cpp:GMainWindow:251: Host Swap: 1.00 GiB
[   0.207751] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:238: Secure1 KeyX missing
[   0.207758] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:248: Secure1 KeyX missing
[   0.207761] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:348: Failed to decrypt
[   0.208116] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:238: Secure1 KeyX missing
[   0.208119] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:248: Secure1 KeyX missing
[   0.208120] Service.FS <Error> core/file_sys/ncch_container.cpp:Load:348: Failed to decrypt
[   0.210363] Frontend <Info> citra_qt/main.cpp:CheckForUpdates:933: Update check started
[   2.277732] Frontend <Info> citra_qt/main.cpp:OnUpdateFound:946: No updates found
[   3.025531] Frontend <Info> citra_qt/main.cpp:BootGame:1164: Citra starting...
[   3.025655] Service.FS <Error> core/file_sys/ncch_container.cpp:LoadHeader:157: 16384
[   3.029521] Frontend <Info> citra_qt/main.cpp:BootGame:1186: Using per game config file for title id 0004000000162F00
[   3.029533] Config <Info> common/settings.cpp:LogSettings:70: Citra Configuration:
[   3.029536] Config <Info> common/settings.cpp:operator():67: Core_UseCpuJit: true
[   3.029536] Config <Info> common/settings.cpp:operator():67: Core_CPUClockPercentage: 100
[   3.029537] Config <Info> common/settings.cpp:operator():67: Renderer_UseGLES: false
[   3.029537] Config <Info> common/settings.cpp:operator():67: Renderer_GraphicsAPI: OpenGL
[   3.029538] Config <Info> common/settings.cpp:operator():67: Renderer_Debug: false
[   3.029538] Config <Info> common/settings.cpp:operator():67: Renderer_UseHwShader: true
[   3.029539] Config <Info> common/settings.cpp:operator():67: Renderer_ShadersAccurateMul: true
[   3.029539] Config <Info> common/settings.cpp:operator():67: Renderer_UseShaderJit: true
[   3.029540] Config <Info> common/settings.cpp:operator():67: Renderer_UseResolutionFactor: 4
[   3.029540] Config <Info> common/settings.cpp:operator():67: Renderer_FrameLimit: 100
[   3.029541] Config <Info> common/settings.cpp:operator():67: Renderer_VSyncNew: true
[   3.029541] Config <Info> common/settings.cpp:operator():67: Renderer_PostProcessingShader: none (builtin)
[   3.029542] Config <Info> common/settings.cpp:operator():67: Renderer_FilterMode: true
[   3.029542] Config <Info> common/settings.cpp:operator():67: Renderer_TextureFilter: None
[   3.029545] Config <Info> common/settings.cpp:operator():67: Stereoscopy_Render3d: 0
[   3.029546] Config <Info> common/settings.cpp:operator():67: Stereoscopy_Factor3d: 0
[   3.029546] Config <Info> common/settings.cpp:operator():67: Stereoscopy_MonoRenderOption: 0
[   3.029547] Config <Info> common/settings.cpp:operator():67: Layout_LayoutOption: 2
[   3.029547] Config <Info> common/settings.cpp:operator():67: Layout_SwapScreen: false
[   3.029548] Config <Info> common/settings.cpp:operator():67: Layout_UprightScreen: false
[   3.029551] Config <Info> common/settings.cpp:operator():67: Layout_LargeScreenProportion: 4
[   3.029552] Config <Info> common/settings.cpp:operator():67: Utility_DumpTextures: false
[   3.029552] Config <Info> common/settings.cpp:operator():67: Utility_CustomTextures: false
[   3.029553] Config <Info> common/settings.cpp:operator():67: Utility_PreloadTextures: false
[   3.029553] Config <Info> common/settings.cpp:operator():67: Utility_AsyncCustomLoading: true
[   3.029554] Config <Info> common/settings.cpp:operator():67: Utility_UseDiskShaderCache: true
[   3.029554] Config <Info> common/settings.cpp:operator():67: Audio_Emulation: HLE
[   3.029555] Config <Info> common/settings.cpp:operator():67: Audio_OutputType: 0
[   3.029556] Config <Info> common/settings.cpp:operator():67: Audio_OutputDevice: auto
[   3.029558] Config <Info> common/settings.cpp:operator():67: Audio_InputType: 0
[   3.029558] Config <Info> common/settings.cpp:operator():67: Audio_InputDevice: auto
[   3.029559] Config <Info> common/settings.cpp:operator():67: Audio_EnableAudioStretching: true
[   3.029559] Config <Info> common/settings.cpp:operator():67: Camera_OuterRightName: blank
[   3.029559] Config <Info> common/settings.cpp:operator():67: Camera_OuterRightConfig: 
[   3.029560] Config <Info> common/settings.cpp:operator():67: Camera_OuterRightFlip: 0
[   3.029560] Config <Info> common/settings.cpp:operator():67: Camera_InnerName: blank
[   3.029561] Config <Info> common/settings.cpp:operator():67: Camera_InnerConfig: 
[   3.029561] Config <Info> common/settings.cpp:operator():67: Camera_InnerFlip: 0
[   3.029563] Config <Info> common/settings.cpp:operator():67: Camera_OuterLeftName: blank
[   3.029563] Config <Info> common/settings.cpp:operator():67: Camera_OuterLeftConfig: 
[   3.029563] Config <Info> common/settings.cpp:operator():67: Camera_OuterLeftFlip: 0
[   3.029564] Config <Info> common/settings.cpp:operator():67: DataStorage_UseVirtualSd: true
[   3.029564] Config <Info> common/settings.cpp:operator():67: DataStorage_UseCustomStorage: false
[   3.029566] Config <Info> common/settings.cpp:operator():67: System_IsNew3ds: true
[   3.029566] Config <Info> common/settings.cpp:operator():67: System_RegionValue: -1
[   3.029568] Config <Info> common/settings.cpp:operator():67: System_PluginLoader: false
[   3.029568] Config <Info> common/settings.cpp:operator():67: System_PluginLoaderAllowed: true
[   3.029569] Config <Info> common/settings.cpp:operator():67: Debugging_UseGdbstub: false
[   3.029569] Config <Info> common/settings.cpp:operator():67: Debugging_GdbstubPort: 24689
[   3.209595] Audio.DSP <Info> audio_core/hle/wmf_decoder.cpp:Impl:67: Media Foundation activated
[   3.210604] Audio.DSP <Info> audio_core/hle/wmf_decoder_utils.cpp:MFDecoderInit:50: Windows(R) Media Foundation found 1 suitable decoder(s)
[   3.255373] Audio.Sink <Info> audio_core/cubeb_sink.cpp:StateCallback:137: Cubeb Audio Stream Started
[   3.255383] RPC_Server <Info> core/rpc/rpc_server.cpp:RPCServer:12: Starting RPC server ...
[   3.255484] RPC_Server <Info> core/rpc/rpc_server.cpp:RPCServer:16: RPC started.
[   3.255535] RPC_Server <Info> core/rpc/rpc_server.cpp:HandleRequestsLoop:114: Request handler started.
[   3.291911] Render.OpenGL <Info> video_core/renderer_opengl/gl_driver.cpp:ReportDriverInfo:134: GL_VERSION: 4.3.0 NVIDIA 528.02
[   3.291915] Render.OpenGL <Info> video_core/renderer_opengl/gl_driver.cpp:ReportDriverInfo:135: GL_VENDOR: NVIDIA Corporation
[   3.291916] Render.OpenGL <Info> video_core/renderer_opengl/gl_driver.cpp:ReportDriverInfo:136: GL_RENDERER: NVIDIA GeForce GTX 1660 Ti/PCIe/SSE2
[   3.318019] Loader <Info> core/loader/ncch.cpp:Load:217: Program ID: 0004000000162F00
[   3.318085] Service.FS <Warning> core/file_sys/ncch_container.cpp:OpenFile:126: Failed to open C:\Users\Raph\AppData\Roaming/Citra/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/title/0004000e/00162f00/content/00000000.app
[   3.349264] Loader <Info> core/hle/kernel/process.cpp:ParseKernelCaps:151: ExHeader kernel version: 2.50
[   3.350672] Service.CFG <Info> core/hle/service/cfg/cfg.cpp:SetPreferredRegionCodes:774: Preferred region code set to 2
[   3.401972] Render.OpenGL <Info> video_core/renderer_opengl/gl_shader_disk_cache.cpp:LoadTransferable:176: Found a transferable disk cache with 2648 entries
[   3.476954] Service.CAM <Info> citra_qt/camera/qt_multimedia_camera.cpp:ResumeCameras:50: Resuming all cameras
[   3.477301] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.01.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477371] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.02.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477427] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.03.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477481] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.04.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477539] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.05.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477591] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.06.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477644] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.07.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477696] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.08.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477796] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.09.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.477885] Core <Warning> core/savestate.cpp:ValidateSaveState:63: Save state file C:\Users\Raph\AppData\Roaming/Citra/states/0004000000162F00.10.cst created from a different revision ca2d87e5e38f84a1f4336c995fa71fb0e235e667
[   3.509467] Render.OpenGL <Info> video_core/renderer_opengl/gl_shader_disk_cache.cpp:LoadPrecompiledFile:279: Found a precompiled disk cache with 2648 decompiled entries and 2648 binary entries
[   4.465143] Service.SRV <Warning> core/hle/service/sm/srv.cpp:RegisterClient:63: (STUBBED) called. Caller PID=11
[   4.515338] Service.SRV <Warning> core/hle/service/sm/srv.cpp:EnableNotification:85: (STUBBED) called
[   4.519851] Service.NDM <Warning> core/hle/service/ndm/ndm_u.cpp:OverrideDefaultDaemons:186: (STUBBED) bit_mask=0x0000000F
[   4.519922] Service.NDM <Warning> core/hle/service/ndm/ndm_u.cpp:SuspendDaemons:74: (STUBBED) bit_mask=0x00000006
[   4.538615] Service.FS <Warning> core/hle/service/fs/fs_user.cpp:InitializeWithSdkVersion:596: (STUBBED) called, version: 0x0B0300C8
[   4.539313] Service.FS <Error> ./core/file_sys/archive_backend.h:GetOpenDelayNs:193: Delay generator was not initalized. Using default
[   4.540920] Service.FS <Error> ./core/file_sys/archive_backend.h:GetOpenDelayNs:193: Delay generator was not initalized. Using default
[   4.555218] Service.SRV <Warning> core/hle/service/sm/srv.cpp:Subscribe:210: (STUBBED) called, notification_id=0x207
[   4.555344] Service.SRV <Warning> core/hle/service/sm/srv.cpp:Subscribe:210: (STUBBED) called, notification_id=0x209
[   4.556611] Service.APT <Warning> core/hle/service/apt/apt.cpp:NotifyToWait:338: (STUBBED) app_id=768
[   4.558849] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:637: (STUBBED) called command=0X00000007, input_size=0X00000004, output_size=0X00000001
[   4.559174] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:637: (STUBBED) called command=0X00000004, input_size=0X00000001, output_size=0X00000001
[   4.559539] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:637: (STUBBED) called command=0X00000004, input_size=0X00000001, output_size=0X00000001
[   4.587620] Service.GSP <Warning> core/hle/service/gsp/gsp_gpu.cpp:AcquireRight:749: called flag=00000000 process=11 thread_id=0
[   4.651689] Service.AC <Warning> core/hle/service/ac/ac.cpp:SetClientVersion:174: (STUBBED) called, version: 0x0B0300C8
[   4.652479] Service.HTTP <Warning> core/hle/service/http_c.cpp:Initialize:126: (STUBBED) called, shared memory size: 0 pid: 11
[   4.653685] Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:165: unknown / unimplemented function 'GetProgramID': port='APT:A' cmd_buf={[0]=0x580002 (0x0058, 0, 2), [1]=0x20, [2]=0xb}
[   4.656406] Service.CFG <Error> core/hle/service/cfg/cfg.cpp:GetCountryCodeID:248: requested country code name=  is invalid
[   4.657876] Service.APT <Warning> core/hle/service/apt/apt.cpp:CheckNew3DSApp:1160: (STUBBED) called
[   4.677257] Service.DSP <Info> audio_core/hle/hle.cpp:LoadComponent:512: Firmware hash: 0xdcdb8c683bf28fdd
[   4.677266] Service.DSP <Info> audio_core/hle/hle.cpp:LoadComponent:516: Structures hash: 0x228655de1258d999
[   4.677268] Service.DSP <Info> core/hle/service/dsp/dsp_dsp.cpp:LoadComponent:192: called size=0xC25C, prog_mask=0x000000FF, data_mask=0x001000FF
[   4.677997] Service.DSP <Info> core/hle/service/dsp/dsp_dsp.cpp:RegisterInterruptEvents:258: Registered interrupt=2, channel=2, event=event-003df570
[   4.678155] Service.DSP <Warning> core/hle/service/dsp/dsp_dsp.cpp:GetSemaphoreEventHandle:274: (STUBBED) called
[   4.678298] Service.DSP <Warning> core/hle/service/dsp/dsp_dsp.cpp:SetSemaphoreMask:284: (STUBBED) called mask=0x2000
[   4.678495] Audio.DSP <Info> audio_core/hle/hle.cpp:PipeWrite:271: Application has requested initialization of DSP hardware
[   4.678598] Service.DSP <Info> core/hle/service/dsp/dsp_dsp.cpp:SetSemaphore:57: called, semaphore_value=4000
[   4.679416] Service.DSP <Info> core/hle/service/dsp/dsp_dsp.cpp:SetSemaphore:57: called, semaphore_value=4000
[   5.572438] Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:165: unknown / unimplemented function 'ControlArchive': port='fs:USER' cmd_buf={[0]=0x80d0144 (0x080D, 5, 4), [1]=0x3, [2]=0x0, [3]=0x0, [4]=0x1, [5]=0x1, [6]=0x1a, [7]=0x0, [8]=0x1c, [9]=0x1}
[   8.203346] Service.FS <Error> ./core/file_sys/archive_backend.h:GetOpenDelayNs:193: Delay generator was not initalized. Using default
[  20.605209] Service.FS <Warning> core/hle/service/fs/file.cpp:OpenLinkFile:214: (STUBBED) File command OpenLinkFile Path: [Binary: 000000000000000000000000]
[  24.246854] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:Initialize:30: called, communication_mode=2
[  24.247063] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:GetActivateEvent:240: called
[  24.247240] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:GetDeactivateEvent:257: called
[  25.280843] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:StartDetection:132: called, in_val=0000
[  41.136990] Service.NFC <Info> core/hle/service/nfc/nfc_device.cpp:LoadAmiibo:112: Loading amiibo with keys
[  41.142241] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:GetTagInfo:331: called
[  41.143262] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:Mount:176: called
[  42.128960] Service.NFC <Info> core/hle/service/nfc/nfc_device.cpp:CloseAmiibo:130: Amiibo removed
[  42.579437] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:Finalize:64: called, communication_mode=2
[  46.259645] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:Initialize:30: called, communication_mode=2
[  46.259658] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:GetActivateEvent:240: called
[  46.259749] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:GetDeactivateEvent:257: called
[  47.293835] Service.NFC <Info> core/hle/service/nfc/nfc.cpp:StartDetection:132: called, in_val=0000
[  51.861794] RPC_Server <Info> core/rpc/rpc_server.cpp:~RPCServer:20: Stopping RPC ...
[  51.862077] RPC_Server <Info> core/rpc/server.cpp:NewRequestCallback:37: Received end packet
[  51.862228] RPC_Server <Info> core/rpc/rpc_server.cpp:~RPCServer:24: RPC stopped.
[  51.874642] Audio.Sink <Info> audio_core/cubeb_sink.cpp:StateCallback:140: Cubeb Audio Stream Stopped
[  51.878941] Kernel <Info> core/hle/kernel/process.cpp:~Process:479: Cleaning up process 11
[  51.878941] Kernel <Info> core/hle/kernel/process.cpp:~Process:479: Cleaning up process 11

### System Configuration

CPU: Intel 12400
GPU/Driver: 1660Ti
RAM: 16gb
OS: Win11

### PR DESCRIPTION
**Pull request by german77**
Chibi robo has a built in delay after mounting the tag to ensure the tag is ready to pull data. We currently remove the tag too soon resulting in an error. Instead of increasing the time to 3-5 seconds I decided to reset the time counter so as long the tag is being constantly used it will never be removed.

I also removed the global usages of system.

Fixes #6681

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6687)

### CODE REVIEW

File: src/core/hle/service/nfc/nfc_device.cpp [hunk 1]
```diff
@@ -1111,4 +1109,18 @@ void NfcDevice::BuildAmiiboWithoutKeys() {
     settings.settings.appdata_initialized.Assign(0);
 }
 
+void NfcDevice::ScheduleTagRemoveEvent() {
```
Comment (SachinVin) [inline src/core/hle/service/nfc/nfc_device.cpp:1112]:
```suggestion
void NfcDevice::RescheduleTagRemoveEvent() {
```
Nit

File: src/core/hle/service/nfc/nfc_device.cpp [hunk 2]
```diff
@@ -1111,4 +1109,18 @@ void NfcDevice::BuildAmiiboWithoutKeys() {
     settings.settings.appdata_initialized.Assign(0);
 }
 
+void NfcDevice::RescheduleTagRemoveEvent() {
+    /// The interval at which the amiibo will be removed automatically 1.5s
+    static constexpr u64 amiibo_removal_interval_ticks = 402 * 1000 * 1000;
```
Comment (Steveice10) [inline src/core/hle/service/nfc/nfc_device.cpp:1114]: * Given the value is different from before, is this still 1.5s?
* Would suggest using ```nsToCycles``` with a time value instead.

Comment (german77) [reply src/core/hle/service/nfc/nfc_device.cpp:1114]: Previously was 1s. Seems like I forgot to correct the comment before.

**Review by Steveice10 — APPROVED** [review]

**Comment by Least100Seraphs** [comment]
Hi,
Just want to comment to confirm that this has resolved issue #6681.

Thanks!

**Pull request merged**